group_consumer.go

```diff
@@ -174,18 +186,20 @@ func (gc *GroupConsumer[T]) consumeLoop() {
 			sendCheckCancel(gc.ctx, gc.consumeChan, Message[T]{Err: ReadError{Err: err}})
 			return
 		case err := <-gc.ackErrChan:
-			if !sendCheckCancel(gc.ctx, gc.consumeChan, ackErrToMessage[T](err, gc.stream)) {
-				gc.lostAcksChan <- err.id
+			if !sendCheckCancel(gc.ctx, gc.consumeChan, ackErrToMessage[T](err, err.Stream)) {
+				gc.lostAcksChan <- err.InnerAck
 			}
 			continue
 		case msg = <-gc.fetchChan:
 		}
 
 		// Eager consume ack messages to keep buffer free and avoid deadlock
-		gc.eagerAckErrorDrain()
+		gc.eagerAckErrorDrain(msg.stream)
 
-		// Send message.
-		sendCheckCancel(gc.ctx, gc.consumeChan, toMessage[T](msg.message, msg.stream))
+		// Send message to consumer.
+		if sendCheckCancel(gc.ctx, gc.consumeChan, toMessage[T](msg.message, msg.stream)) {
+			fetchedIds[msg.stream] = msg.message.ID
+		}
```

Comment:
IIRC the fetched (previously seen) ids store what we already fetched from Redis, not what we provided to the consumer channel (I mean after all this line wasn't there before)